be/src/env/env_s3.cpp

```diff
@@ -35,41 +30,33 @@ class S3RandomAccessFile : public RandomAccessFile {
     const std::string& file_name() const override { return _object_path; }
 
 private:
-    std::unique_ptr<ObjectStore> _client;
-    std::string _bucket;
-    std::string _object;
+    std::unique_ptr<io::RandomAccessFile> _input;
     std::string _object_path;
     mutable uint64_t _object_size;
 };
 
-Status S3RandomAccessFile::read(uint64_t offset, Slice* res) const {
-    size_t read_size = 0;
-    RETURN_IF_ERROR(_client->get_object_range(_bucket, _object, offset, res->size, res->data, &read_size));
-    res->size = read_size;
+Status RandomAccessFileWrapper::read(uint64_t offset, Slice* res) const {
+    if (UNLIKELY(offset > std::numeric_limits<int64_t>::max())) return Status::NotSupported("offset overflow");
```

Comment:
this seems will never be true.

Reply:
Do you mean this won't happen in practice?

Reply:
if `offset` == `std::numeric_limits<int64_t>::max()`, i think `offset + 1` will be 0?

Reply:
i see, it's int64_t not uint64_t.

Reply:
Yes, you're right!